src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/ResourceServerConfiguration.kt

```diff
@@ -1,62 +1,45 @@
 package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.config
 
-import org.springframework.beans.factory.annotation.Value
-import org.springframework.cache.CacheManager
 import org.springframework.cache.annotation.EnableCaching
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.data.web.config.EnableSpringDataWebSupport
 import org.springframework.data.web.config.EnableSpringDataWebSupport.PageSerializationMode.VIA_DTO
 import org.springframework.scheduling.annotation.EnableAsync
-import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity
 import org.springframework.security.config.annotation.web.builders.HttpSecurity
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
-import org.springframework.security.config.annotation.web.invoke
-import org.springframework.security.config.http.SessionCreationPolicy
-import org.springframework.security.oauth2.jwt.JwtDecoder
-import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
 import org.springframework.security.web.SecurityFilterChain
+import uk.gov.justice.hmpps.kotlin.auth.HmppsResourceServerConfiguration
+import uk.gov.justice.hmpps.kotlin.auth.dsl.ResourceServerConfigurationCustomizer
 
 @Configuration
-@EnableWebSecurity
-@EnableMethodSecurity(prePostEnabled = true)
 @EnableCaching
 @EnableAsync
 @EnableSpringDataWebSupport(pageSerializationMode = VIA_DTO)
 class ResourceServerConfiguration {
 
   @Bean
-  fun filterChain(http: HttpSecurity): SecurityFilterChain {
-    http {
-      headers { frameOptions { sameOrigin = true } }
-      sessionManagement { sessionCreationPolicy = SessionCreationPolicy.STATELESS }
-      // Can't have CSRF protection as requires session
-      csrf { disable() }
-      authorizeHttpRequests {
-        listOf(
-          "/webjars/**",
-          "/favicon.ico",
-          "/health/**",
-          "/info",
-          "/h2-console/**",
-          "/v3/api-docs/**",
-          "/swagger-ui.html",
-          "/swagger-ui/**",
-          "/swagger-resources",
-          "/swagger-resources/configuration/ui",
-          "/swagger-resources/configuration/security",
-          "/jobs/**",
```

Comment:
we'll need to add `jobs` as an allowed list.. probably worth checking the others just to be sure